client/src/components/useFileUpload.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  useQuery,
+  useMutation,
+  useQueryClient,
+  UseQueryResult,
+} from 'react-query'
+import axios, { AxiosRequestConfig } from 'axios'
+import { useState, useRef } from 'react'
+import { IFileInfo, CvrFileType } from './useCSV'
+import { fetchApi, ApiError, addCSRFToken } from '../utils/api'
+
+interface IUseFileUploadProps {
+  url: string
+  formKey: string
+}
+
+export interface IFileUpload {
+  uploadedFile: UseQueryResult<IFileInfo, ApiError>
+  uploadFiles: (files: File[]) => Promise<void>
+  uploadProgress?: number
+  deleteFile: () => Promise<void>
+  downloadFileUrl?: string
+}
+
+export const useUploadedFile = (
+  key: string[],
+  url: string,
+  options: { onFileChange?: () => void; enabled?: boolean } = {}
+) => {
+  const isFirstFetch = useRef(true)
+  const isProcessing = (fileInfo?: IFileInfo) =>
+    fileInfo && fileInfo.processing && !fileInfo.processing.completedAt
+  return useQuery<IFileInfo, ApiError>(key, () => fetchApi(url), {
+    // When a file input dialog closes, it triggers a window focus event,
+    // which causes a refetch by default, so we turn that off.
+    // https://github.com/tannerlinsley/react-query/issues/2960
+    refetchOnWindowFocus: false,
```

Comment:
Yeah, I thought about that! I think refetching on window focus could be nice for users in general (imagining if they switch tabs and come back). That being said, it's definitely a nice-to-have and Arlo's been totally fine without it for now. What do you think? I could go either way.

Reply:
Good point about refetching when users switch tabs and come back!

I also don't have a strong preference here but like the idea of making `refetchOnWindowFocus: false` the default to avoid confusion when writing tests involving interactions that trigger window focus (e.g. "Why is this fetch call happening twice when I think it should be happening once? Has my code negatively impacted client-side caching?")

If we do do this, I realize some (many?) existing tests may require updating so feel free to defer to a follow-up issue!

Reply:
I like that reasoning. I'll try making it the default and see how it goes

Reply:
Ok it was pretty easy - just had to change one test